build_test.go

```diff
@@ -1343,168 +1275,285 @@ func testBuild(t *testing.T, when spec.G, it spec.S) {
+								Image:        "some/app",
+								Builder:      defaultBuilderName,
+								Publish:      true,
+								TrustBuilder: true,
+							}))
+							h.AssertEq(t, fakeLifecycle.Opts.UseCreator, false)
+							h.AssertEq(t, fakeLifecycle.Opts.LifecycleImage, defaultBuilderImage.Name())
+
+							args := fakeImageFetcher.FetchCalls[fakeLifecycleImage.Name()]
+							h.AssertNil(t, args)
+						})
 					})
 				})
 			})
 
-			when("NoPull option", func() {
-				when("true", func() {
-					it("uses the local builder and run images without updating", func() {
+			when("false", func() {
+				it("uses a local run image", func() {
+					h.AssertNil(t, subject.Build(context.TODO(), BuildOptions{
+						Image:   "some/app",
+						Builder: defaultBuilderName,
+						Publish: false,
+					}))
+					h.AssertEq(t, fakeLifecycle.Opts.Publish, false)
+
+					args := fakeImageFetcher.FetchCalls["default/run"]
+					h.AssertEq(t, args.Daemon, true)
+					h.AssertEq(t, args.Pull, true)
+
+					args = fakeImageFetcher.FetchCalls[defaultBuilderName]
+					h.AssertEq(t, args.Daemon, true)
+					h.AssertEq(t, args.Pull, true)
+				})
+
+				when("lifecycle supports creator", func() {
+					it("uses the creator", func() {
 						h.AssertNil(t, subject.Build(context.TODO(), BuildOptions{
-							Image:   "some/app",
-							Builder: builderName,
-							NoPull:  true,
+							Image:        "some/app",
+							Builder:      builderImageSupportingCreator.Name(),
+							Publish:      false,
+							TrustBuilder: false,
 						}))
+						h.AssertEq(t, fakeLifecycle.Opts.UseCreator, true)
 
-						args := fakeImageFetcher.FetchCalls["default/run"]
-						h.AssertEq(t, args.Daemon, true)
-						h.AssertEq(t, args.Pull, false)
-
-						args = fakeImageFetcher.FetchCalls[builderName]
-						h.AssertEq(t, args.Daemon, true)
-						h.AssertEq(t, args.Pull, false)
+						args := fakeImageFetcher.FetchCalls[fakeLifecycleImage.Name()]
+						h.AssertNil(t, args)
 					})
 				})
 
-				when("false", func() {
-					it("uses pulls the builder and run image before using them", func() {
+				when("lifecycle doesn't support creator", func() {
+					// the default test builder (example.com/default/builder:tag) has lifecycle version 0.3.0, so creator is not supported
+					it("uses the 5 phases with the provided builder", func() {
 						h.AssertNil(t, subject.Build(context.TODO(), BuildOptions{
-							Image:   "some/app",
-							Builder: builderName,
-							NoPull:  false,
+							Image:        "some/app",
+							Builder:      defaultBuilderName,
+							Publish:      false,
+							TrustBuilder: true,
 						}))
+						h.AssertEq(t, fakeLifecycle.Opts.UseCreator, false)
+						h.AssertEq(t, fakeLifecycle.Opts.LifecycleImage, defaultBuilderImage.Name())
+					})
+				})
+			})
+		})
 
-						args := fakeImageFetcher.FetchCalls["default/run"]
-						h.AssertEq(t, args.Daemon, true)
-						h.AssertEq(t, args.Pull, true)
+		when("NoPull option", func() {
```

Comment:
The groupings were messed up in this file - all tests below were nested under this group.